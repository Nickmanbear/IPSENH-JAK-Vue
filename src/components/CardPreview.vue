<template>
  <div>
    <div id="cardprev" v-on:click="toggleCard">
      <p>{{ card.name }}</p>
    </div>
    <Backdrop v-show="show" v-on:clicked="toggleCard"/>
    <Card v-show="show" v-bind:card="card"/>
  </div>
</template>

<script>
import Card from '@/components/Card.vue';
import Backdrop from '@/components/UI/Backdrop.vue';

export default {
  name: 'Home',
  components: {
    Backdrop,
    Card,
  },
  props: {
    card: Object,
  },
  data() {
    return {
      show: false,
    };
  },
  methods: {
    toggleCard() {
      this.show = !this.show;
    },
  },
};
</script>

<style lang="scss">
  #cardprev {
    background-color: #fff;
    border: 1px solid #fff;
    border-radius: 3px;
    padding: 0 8px;
    margin: 8px 0;
    box-shadow: 1px 1px 2px -1px rgba(0,0,0,0.5);
  }

  #cardprev:hover {
    cursor: pointer;
    background-color: #fbfbfb;
    border: 1px solid #ddd;
    box-shadow: none;


  }
</style>
