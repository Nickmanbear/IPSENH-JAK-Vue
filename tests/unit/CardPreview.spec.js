import { shallowMount } from '@vue/test-utils';
import CardPreview from '@/components/CardPreview.vue';

jest.mock('axios', () => ({
  create: () => ({ delete: () => Promise.resolve() }),
}));

describe('CardPreview', () => {
  let cmp;

  beforeEach(() => {
    cmp = shallowMount(CardPreview, {
      propsData: {
        card: {
          id: 5,
          columnId: 1,
          name: 'test card',
          description: 'test description for the card'
        },
      },
    });
  });

  it('should toggle the card modal', () => {
    cmp.vm.toggleCard();
    expect(cmp.vm.show).toBeTruthy();
  });

  it('should delete itself', async () => {
    await cmp.vm.deleteCard();
    expect(cmp.emitted().deleted).toBeTruthy();
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
