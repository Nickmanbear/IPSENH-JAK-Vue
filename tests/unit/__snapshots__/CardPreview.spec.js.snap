// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPreview has the expected html structure 1`] = `
<div>
  <div
    id="cardprev"
  >
    <p>
      test card
    </p>
     
    <button
      id="delete"
    >
      Ã—
    </button>
  </div>
   
  <!---->
   
  <!---->
</div>
`;
